### PR TITLE
[master] Add Linode `backups_enabled` Option; Improve Linode Cloud Docs

### LINKED ISSUE #65697: [FEATURE REQUEST] Backup Enabled for Linode Provisioned from Salt Cloud
It would be nice to have `backups_enabled` option in the Linode module of Salt Cloud to allow the instances created from Salt Cloud to enable Linode auto backup feature.

### PR DESCRIPTION
**Pull request by zliang-akamai**
### What does this PR do?

This PR does:
- improved documentation for Linode module of Salt Cloud
- Add `backups_enabled` option for Linode instance creation

### What issues does this PR fix or reference?
Fixes: #65697

### Previous Behavior
`backups_enabled` option doesn't exist.

### New Behavior
Users can now configure `backups_enabled` in the Linode profile/provider

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Review by dwoz — APPROVED** [review]

**Review by s0undt3ch — APPROVED** [review]

**Comment by dwoz** [comment]
I'll merge this after the merge forward from 3006.5 lands.

**Pull request merged**
